Add new field flow

rijkvanzanten opened the pull request:
Adds a new simplified field flow for field creation. The current flow is persisted for advanced editing, and field updating.

https://user-images.githubusercontent.com/9141017/138619669-7ac7fc37-c858-42cf-91d0-00b1151b1d34.mp4

This PR completely replaces the underlying store that does all the "magic" that controls the various data points that are generated. 

## To-Do:

- [x] Update remaining interfaces with custom options vue components
- [x] Require relationship to be set before save
- [x] Set localType on start editing existing item
- [x] Responsiveness
- [x] Checkout advanced vs regular interface options
- [x] Remove note from simple setup
- [x] Add divider between schema and interface options in simple
- [x] Use subdued background in simple setup
- [x] Add required asterisk to key
- [x] Update points made below

rijkvanzanten committed:
Add new grouping setup to interfaces

rijkvanzanten committed:
[WIP] Start on new field flow

rijkvanzanten committed:
Merge branch 'main' into feat/field-flow

rijkvanzanten committed:
Add preview svgs

rijkvanzanten committed:
Update preview svg

rijkvanzanten committed:
Add basic saving mechanism. Start on magic

rijkvanzanten committed:
Add interface options to simple inputs

rijkvanzanten committed:
Fix missing t

rijkvanzanten committed:
Fix custom options not rendering

rijkvanzanten committed:
Fix interface options overrides

rijkvanzanten committed:
Sync m2o name key

rijkvanzanten committed:
Setup m2o related collection input

rijkvanzanten committed:
Add collections generation

rijkvanzanten committed:
Add circular prevention & type syncing

rijkvanzanten committed:
Small fixes in m2o context generation

rijkvanzanten committed:
Move o2m options to fn structure

rijkvanzanten committed:
Conclude o2m (i think)

rijkvanzanten committed:
Start on toggle to advanced

rijkvanzanten committed:
Style advanced toggle button

rijkvanzanten committed:
Add localType File support, start documenting

rijkvanzanten committed:
Add presentation/group localtypes

rijkvanzanten committed:
Merge branch 'main' into feat/field-flow

rijkvanzanten committed:
Use function style interface options in m2m

rijkvanzanten committed:
Don't require collection prop in field template component

rijkvanzanten committed:
Implement most of m2m magic

rijkvanzanten committed:
Restructure store updater

rijkvanzanten committed:
Finish useFieldDetailStore restructure

rijkvanzanten committed:
Only register root index in module folders

rijkvanzanten committed:
Do the thing

rijkvanzanten committed:
Finish m2m magic, tweak fallback styles

rijkvanzanten committed:
Fix o2m display template setup

rijkvanzanten committed:
Remove options comp for translations interface

rijkvanzanten committed:
Show languages collection picker in translations

rijkvanzanten committed:
Finish translations alterations

rijkvanzanten committed:
Add magic for files type

rijkvanzanten committed:
Officially wave goodbye to the old store setup

It served it's purpose, but what a nightmare to maintain. Welcome to the new version

rijkvanzanten committed:
Update schema tab

rijkvanzanten committed:
Add field pane

rijkvanzanten committed:
Finish interface section

rijkvanzanten committed:
Add display section

rijkvanzanten committed:
Add conditions

rijkvanzanten committed:
Update tabs to use localType from store

rijkvanzanten committed:
Start on relationship tab

rijkvanzanten committed:
Update m2m relational setup

rijkvanzanten committed:
Start on m2o

rijkvanzanten committed:
Finish m2o setup

rijkvanzanten committed:
Finish o2m setup

rijkvanzanten committed:
Add m2a magic

rijkvanzanten committed:
Various tweaks

rijkvanzanten committed:
Add m2a setup

rijkvanzanten committed:
Add save button to advanced flow

rijkvanzanten committed:
Load existing values on start

rijkvanzanten committed:
Add upsert to stores, allow updating existing items

rijkvanzanten committed:
Please the linter gods

azrikahar reviewed (commented):
Findings from test:

- [ ] When we halfway configured a field (filled up key, default value etc) then unclick it to click another field, the values are still filled up with previous values. Maybe resetting it is better, but then when we want to go back to the field we abandoned, not resetting is better. 
- [x] Not sure should we hide the Display Template when Related Collection is empty or change the warning, since technically it's "not yet configured" instead of misconfigured. Filling up Related Collection does make Display Template work as usual.
- [x] Related to above Related Collection (or m2m, o2m, m2o etc), we can save the field without configuring Related Collection, which shouldn't be allowed.
- [x] Clicking Save when creating Groups (any of the 3) seems to freeze or trigger `Maximum Call Stack Size Exceeded` error. However after refreshing, they do seem to be created and work properly.

- ~~Hash field: clicking Save triggers error `Cannot read properties of null (reading 'emitsOptions')`. After refreshing, still gets created & works.~~ Doesn't seem to happen anymore... Might be an unrelated error after the freeze above.

rijkvanzanten committed:
Remove seemingly redundant interface option

@Oreilles Seeing the geometry type is already configured in the schema configuration, this particular option on the interface feels redundant (?)

rijkvanzanten committed:
Remove unnecessary option component overrides

rijkvanzanten committed:
Track relationships in required fields for save state

rijkvanzanten committed:
Fix relations previews on editing existing fields

rijkvanzanten committed:
Use standard vs advanced for input options

rijkvanzanten committed:
Remove note from simple field setup

rijkvanzanten committed:
Add divider to field config

rijkvanzanten committed:
Use background subdued

rijkvanzanten committed:
Default required to false

rijkvanzanten committed:
Add required icons to key/relationship

rijkvanzanten committed:
Tweak colors

rijkvanzanten committed:
Fix infinite loop in group creation

rijkvanzanten committed:
Make setup responsive

rijkvanzanten committed:
Allow switching interface at will

rijkvanzanten committed:
Add m2a related collections picker

rijkvanzanten committed:
Fix relations persisting on field deletion

rijkvanzanten committed:
Add "Create in Advanced" shortcut

Just for you @joselcvarela

rijkvanzanten committed:
Fix PK field staging on collection create

rijkvanzanten committed:
Revert "Remove seemingly redundant interface option"

This reverts commit e5e09a051e76795436e0324540b194c99a743874.

rijkvanzanten committed:
Fix map interface options

rijkvanzanten committed:
Merge branch 'main' into feat/field-flow
